doc/design/design_model_zoo.md

```diff
@@ -0,0 +1,187 @@
+        // model. If the input data does not fit the definitions under "columns" section,
+        // the error will be raised.
+        "columns": {
+            "col1": {
+                "Shape": [1],
+                "IsSparse": false,
+                "Delimiter": "",
+                "DType": "Float",
+            },
+            "col2": {...},
+        },
+        "label": {
+            "Shape": [1],
+                "IsSparse": false,
+                "Delimiter": "",
+                "DType": "Int",
+        },
+    }
+   ```
+- `model_def.py` file is the model python definition. The content varies when using different engines:
+    - Custom Estimator: A sub class of `tf.estimator.Estimator`
+    - Keras Model: A keras sub class model definition.
+    - XGBoost Model: One line indicating the XGBoost supported model type, like: `model_type = xgboost.gbtree`
+
+## ElasticDL Compatible Model
+
+On the one hand, the model trained by ElasticDL can also be published in SQLFlow model zoo. A `model_meta.json` file will be generated when training ElasticDL.
+
+On the other hand, If you want to train/fine-tune a model from the model zoo with ElasticDL, SQLFlow can use the "columns" information in `model_meta.json` to form a `dataset_fn` which ElasticDL needed.
+see: https://github.com/sql-machine-learning/sqlflow/blob/develop/pkg/sql/template_elasticdl.go#L46
+
+## Publish A Model to the Model Zoo
+
+To publish a new model into the model zoo, you need to:
+
+1. commit your model (code and README.md) to https://github.com/sql-machine-learning/models and merge the code. A model in the models repo should contain below files:
+    ```
+    - README.md
+    - requirements.txt
+    - model_def.py
+    - some_deps.py
+    ```
+1. train and test your model on the dataset then save the model weights together with `model_meta.json` file.
+1. name a directory and upload all files listed above into the directory on the server.
+
+The overall workflow to publish and use a model in the model zoo is shown below:
+
+<p align="center">
+<img src="../figures/modelzoo_workflow.png" width=500px>
+</p>
+
+## Model Sharing
+
+Model sharing is a necessary feature to encourage more users to contribute models to the model zoo. The
+features are quite common for products like [DockerHub](https://hub.docker.com/). We can discuss these
+features when we need to implement the model sharing features.
+
+## Use the Model Zoo in SQLFlow Statements
+
+In SQLFlow, you can specify the model under `sqlflow.org/modelzoo` in the `TRAIN` and `USING` clause.
+
+If the `TRAIN` clause accepts a model under `sqlflow.org/modelzoo`, SQLFlow will only use the model definition to start the train. If `USING` clause accepts a model under `sqlflow.org/modelzoo`, the model's
+weights will be loaded both in `TRAIN` process or `PREDICT` process.
+
+We can use the models in the model zoo to start below jobs in minutes:
+
+1. Predict some data
+1. Transfer learning to fit your data
```

Comment:
We can name it refit instead of transfer learning. It seems that we didn't change the IO and network architecture of this model.